fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/StoragePreconditions.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.storage;
+
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Preconditions that can be used when working with storages.
```

Comment:
If you want to ignore my really petty grammar suggestions that's fine, but why not set up code examples in all of the utility methods like this one? it may take some time and might be better in a subsequent PR but it would be nice to have some general javadocs on how this should be used in general AND/OR just a bit more of where exactly this ties into.

Reply:
Added a remark, and an example for `StorageUtil#move`.